drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -375,11 +460,27 @@ static int sdw_compute_bus_params(struct sdw_bus *bus)
 				(bus->params.max_dr_freq >>  clk_buf[i]) :
 				clk_buf[i] * SDW_DOUBLE_RATE_FACTOR;
 
-		if (curr_dr_freq * (mstr_prop->default_col - 1) <
+		if (curr_dr_freq * (mstr_prop->default_col - 1) >=
 		    bus->params.bandwidth * mstr_prop->default_col)
-			continue;
+			break;
+
+		list_for_each_entry(m_rt, &bus->m_rt_list, bus_node) {
+			/*
+			 * Get the first s_rt that will be used to find the available lane that
+			 * can be used.
+			 */
+			s_rt = list_first_entry(&m_rt->slave_rt_list,
+						struct sdw_slave_runtime, m_rt_node);
 
-		break;
+			/*
+			 * Find available manager lanes that connected to the first Peripheral.
+			 * No need to check all Peripherals available lanes because we can't use
+			 * multi-lane if we can't find any available lane for the first Peripheral.
```

Comment:
get_manager_lane() will check all peripherals lane connection via a call to is_lane_connected_to_all_peripherals().
Isn't this a bit contradicting to this comment?

Reply:
That means something different. "No need to check all Peripherals" here means we only need to check the available lanes of the first Peripheral. Then, `is_lane_connected_to_all_peripherals()` will check if the lane can be used for all Peripherals. We don't need to repeat checking all Peripherals because one lane can not be used if the lane is not connected to the first Peripheral.

Reply:
Reword it. Hope it is more readable. :)